addons/l10n_it_edi_sdicoop/models/account_edi_format.py

```diff
@@ -163,14 +164,16 @@ def _l10n_it_post_invoices_step_1(self, invoices):
                 'error': _("You must accept the terms and conditions in the settings to use FatturaPA."),
                 'blocking_level': 'error'} for invoice in invoices}
 
+        responses = {}
         if proxy_user._get_demo_state() == 'demo':
-            responses = {filename: {'id_transaction': 'demo'} for invoice in invoices}
-        else:
+            responses = {i['data']['filename']: {'id_transaction': 'demo'} for i in to_send.values()}
```

Comment:
If this was causing a traceback, it would be nice to test it too in my opinion. Better be safe :smile:

Reply:
Should this not be done in `_l10n_it_edi_upload` to avoid the `elif to_send:` and to make the data structure come from the same function?

Reply:
We can do that, but I find it counterintuitive as upload method seems to suppose it is doing the call.

Reply:
> as upload method seems to suppose it is doing the call.

What?

Reply:
It could be weird to call a method upload that than does not really upload anything. Anyways, now that method will return {} before doing the call if there is nothing in there, so we can keep the else:

Reply:
whatever...